### PR TITLE
feat: support anonymous company reviews 👻 

### LINKED ISSUE #302: Ability to post anonymous company review 👻
### Description

Members should have the ability to post a company review anonymously. On the Companies page, on the modal that appears after pressing "Add Review", there should be a checkbox with a description that specifies when selected, review will be posted anonymously. The default should not be anonymous posting, but this feature will allow members to opt into anonymously reviewing if they prefer. 

### Acceptance Criteria

- [ ] Add a BOOLEAN column `anonymous` to the database table `company_review` through a database migration. The default of `anonymous` should be `false`
- [ ] On the "Add Review" modal [here](https://github.com/colorstackorg/oyster/blob/main/apps/member-profile/app/shared/components/review-form.tsx), include a checkbox with a description 
- [ ] The state of this checkbox should reflect the current `anonymous` value for the review (checked if true)
- [ ] Update the `anonymous` column in the database when the checkbox is toggled.
- [ ] When company reviews are displayed, if `anonymous` for a given review is `true`, the name for the reviewer should be "Anonymous", and there should be a blank profile picture

### Additional Context

Check out the `checkbox` component [here](https://github.com/colorstackorg/oyster/blob/main/packages/ui/src/components/checkbox.tsx#L22-L79), and see examples of database migrations [here!](https://github.com/colorstackorg/oyster/tree/main/packages/db/src/migrations)
Feel free to add or ask for any more insight!

### PR DESCRIPTION
**Pull request by jasonappah**
## Description ✏️

Closes #302.

This PR adds support for anonymous company reviews, where members can create company reviews without their names being displayed.
CleanShot 2024-08-08 at 17 14 22@2x
CleanShot 2024-08-08 at 17 13 50@2x

## Type of Change 🐞

- [x] Feature - A non-breaking change which adds functionality.
- [ ] Fix - A non-breaking change which fixes an issue.
- [ ] Refactor - A change that neither fixes a bug nor adds a feature.
- [ ] Documentation - A change only to in-code or markdown documentation.
- [ ] Tests - A change that adds missing unit/integration tests.
- [ ] Chore - A change that is likely none of the above.

## Checklist ✅

- [x] I have done a self-review of my code.
- [x] I have manually tested my code (if applicable).
- [ ] I have added/updated any relevant documentation (if applicable).

### CODE REVIEW

File: apps/member-profile/app/shared/components/review-form.tsx [hunk 1]
```diff
@@ -225,6 +235,18 @@ function RecommendField({ defaultValue, error, name }: FieldProps<boolean>) {
   );
 }
 
+function AnonymousField({ error, name, defaultValue }: FieldProps<boolean>) {
+  return (
+    <Form.Field
+      error={error}
+      label="Would you like to remain anonymous?"
```
Comment (tomas-salgado) [inline apps/member-profile/app/shared/components/review-form.tsx:242]: nit: we could change this to `Would you like to post this review anonymously?`

**Review by tomas-salgado — COMMENTED** [review]
Good work @jasonappah! This PR is almost there, just a few things to add. 

I tried testing the feature locally, but it seems that the `anonymous` field is not getting updated in the database when I create a new review. I suspect the issue here is that we need to pass in `anonymous` in [add-company-review.ts](https://github.com/colorstackorg/oyster/blob/main/packages/core/src/modules/employment/use-cases/add-company-review.ts#L13), and similarly in [edit-company-review.ts](https://github.com/colorstackorg/oyster/blob/82316f5b7198725acbf5df0212480e12307a3258/packages/core/src/modules/employment/use-cases/edit-company-review.ts). Then we just need to ensure that we're loading this value from the database properly. 

Let me know if you need any help on this or have any questions/comments!

File: apps/member-profile/app/shared/components/company-review.tsx [hunk 2]
```diff
@@ -67,17 +68,24 @@ export const CompanyReview = ({
...
       <header className="flex items-center gap-1">
-        <CompanyReviewer
-          reviewerFirstName={reviewerFirstName}
-          reviewerLastName={reviewerLastName}
-          reviewerId={reviewerId}
-          reviewerProfilePicture={reviewerProfilePicture}
-        />
+        {anonymous ? (
+          <Text color="gray-500" variant="sm">
+            Anonymous Member
```
[the remark this answers is not in the document]
Comment (jasonappah) [reply apps/member-profile/app/shared/components/company-review.tsx]: fixed in `49cf515aa4c266e18984609d1c5a04b01488dc5a`

File: apps/member-profile/app/shared/components/review-form.tsx (hunk 1 above)
Comment (jasonappah) [reply apps/member-profile/app/shared/components/review-form.tsx]: fixed in `49cf515aa4c266e18984609d1c5a04b01488dc5a`

**Comment by jasonappah** [comment]
Tested again locally and everything seems to be good.

**Review by ramiAbdou — APPROVED** [review]
Looks great, @jasonappah! Thank you so much for the work here!

I'm expecting some 2/10 reviews to be posted anonymously soon! 😂😂

**Pull request merged**